docs/docs.logflare.com/docs/self-hosting/index.md

```diff
@@ -56,6 +58,25 @@ Additional environment variable configurations for the OpenTelemetry libraries u
 | `POSTGRES_BACKEND_URL`    | string, required                       | PostgreSQL connection string, for connecting to the database. User must have sufficient permssions to manage the schema. |
 | `POSTGRES_BACKEND_SCHEMA` | string, optional, defaults to `public` | Specifies the database schema to scope all operations.                                                                   |
 
+## Database Encryption
+
+Certain database columns that store sensitive data are encrypted with the `LOGFLARE_DB_ENCRYPTION_KEY` key.
+Encryption keys must be Base64 encoded.
+
+Cipher used is AES with a 256-bit key in GCM mode.
+
+### Rolling Encryption Keys
+
+In order to roll encryption keys and migrate existing encrypted data, use the `LOGFLARE_DB_ENCRYPTION_KEY_RETIRED` environment variable.
+
+Steps to perform the migration are:
+
+1. Move the retired encryption key from `LOGFLARE_DB_ENCRYPTION_KEY` to `LOGFLARE_DB_ENCRYPTION_KEY_RETIRED`.
+2. Generate a new encryption key and set it to `LOGFLARE_DB_ENCRYPTION_KEY`.
+3. Restart or deploy the server with the new environment variables.
+4. Upon successful server startup, an `info` log will be emitted that says that an retired encryption key is detected, and the migration will be initiated to transition all data encrypted with the retired key to be encrypted with the new key.
+5. Once the migration is complete, the retired encryption key can be safely removed.
```

Comment:
"obvious" question, is there something that indicates migration is complete?

Reply:
yea we'll need to add a log line for it, good catch 👍